### PR TITLE
30/60/144 FPS IN SS13, THIS IS NOT A PRANK

### PR DESCRIPTION
**Pull request by Iamgoofball**
AAAAAA

### CODE REVIEW

**Comment by Iamgoofball** [comment]
FANCY BUTTON TOO LOOKIE
dreamseeker_2016-07-14_03-55-23

**Comment by Aloraydrel** [comment]
The human eye cannot see past 10 fps

**Comment by Iamgoofball** [comment]
CINEMATIC

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -941,5 +941,14 @@ var/next_mob_id = 0
...
 /mob/proc/get_idcard()
 	return
+
+/mob/verb/fpssetting(fps as num)
+	if(fps <= 144 && fps >= 10)
+		client.fps = fps
+		src << "Your Frames Per Second(FPS) has been set to [fps]."
+	else
+		src << "Your FPS must be between 10 and 144."
+	return
```
Comment (MrPerson) [inline code/modules/mob/mob.dm]: Before remie gets on your case, trailing return

Also just do fps = Clamp(fps, 10, 144); src << "Your FPS has been set to [fps]."

File: interface/skin.dmf [hunk 2]
```diff
@@ -1,880 +1,923 @@
...
+	elem "set-fps"
+		type = BUTTON
+		pos = 10,4
+		size = 89x20
+		anchor1 = none
+		anchor2 = none
+		background-color = #e9f4f8
+		saved-params = "is-checked"
+		text = "Set FPS"
+		command = "fpssetting"
```
Comment (MrPerson) [inline interface/skin.dmf]: You were warned about the mess this would cause and I'll murder you if you don't fix this

Comment (Iamgoofball) [inline interface/skin.dmf]: i will this was a hack job done by the CIA to rush it out to show MSO before he passes out

File: code/modules/mob/mob.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/modules/mob/mob.dm]: ok

**Comment by Incoming5643** [comment]
S T O P

**Comment by MrStonedOne** [comment]
\>incoming not reading/understanding a pr before commenting

for shame

**Comment by lzimann** [comment]
so, since goof is shit in making pr descriptions, this is client-side fps change, right?

**Comment by AnturK** [comment]
Yes, this adds a toggle for the new byond 511 variable.

**Comment by Incoming5643** [comment]
@MrStonedOne I didn't misread it. I'm just worried that "oh clients can set their fps down if they have a problem!" Will lead to stupid fast server side fps and/or put me at some sort of disadvantage in terms of reaction times and/or still somehow overtax my toaster even if I set the fps back down to 20.

**Comment by ChangelingRain** [comment]
Goof, don't remove the singularity's animations, they don't *just* spin, they have other animations than just that(this is really visible with the supermatter-charged singularity).
At least split it up into another PR so it can be denied properly.

**Comment by Firecage** [comment]
Oh, so the PR has a stealth change which isn't even supposed to be part of this PR?

**Comment by MrStonedOne** [comment]
oh no, its suppose to be a part of this pr @Firecage 

When everything is working in 60fps, the sing rotating at 11fps because it uses sprite based rotations looked super lame and out of place.

if somebody like wjohn wanted to fix the animation to just have the other stuff, no rotation, they could at any time after this is merged.

**Comment by Kromgar** [comment]
So what does the singulo look like now? Is nar'sie no logner going to have animation what about ratvar

**Comment by MrStonedOne** [comment]
narnar and ratvar are unchanges, the sing doesn't have some of its animation, its now just a static icon that rotates via the same command admins use to make something spin, meaning it's rotation is client side and made smoother by higher client side fps.

**Comment by ChangelingRain** [comment]
I'd rather not have this merged unless the singularities retain their non-spin animations, honestly. It should almost certainly be its own pr anyway.

**Comment by MrStonedOne** [comment]
its related to client fps.

this also changes pixel projectiles for client fps reasons (slightly broke them, but that can fix in post)

This is for the 511 branch so we'll have 2 weeks or so before that's stable enough to get merged anyways.

worst case it's reverted before 511branch is merged to master

**Comment by Iamgoofball** [comment]
>I'd rather not have this merged unless the singularities retain their non-spin animations, honestly. It should almost certainly be its own pr anyway.

It looks really nice though with the new 60 FPS shit, have you looked at it on the test server?

**Comment by Iamgoofball** [comment]
>Oh, so the PR has a stealth change which isn't even supposed to be part of this PR?

also fuck off this is literally our test feature for the FPS as it's a good demo

it also shaves 2 MB off the download

**Comment by MrStonedOne** [comment]
@WJohn if you had the spare time, we could use a new sprite for the singulo, that has no rotation, but has the other effects.

**Comment by Iamgoofball** [comment]
lost the election, campaign PRs closing down

**Comment by Iamgoofball** [comment]
https://www.youtube.com/watch?v=_wldE_4xjVQ

**Comment by as334** [comment]
Oh good, I need this for high FPS in my SS13 VR.

**Comment by hornygranny** [comment]
this is cool but because of the way our camera snaps to tiles it doesn't really make movement smooth if you're at sanic speed (which normal spacemen are)

**Comment by MrStonedOne** [comment]
hg, it's more that our glide_size is statically locked to 8px per server fps, regardless of actual movement speed.

I'll try making another smooth movement pr, maybe with client side fps it will fair better.

**Comment by optimumtact** [comment]
goof unbork your shit

**Comment by Iamgoofball** [comment]
were still doing rnd

**Comment by optimumtact** [comment]
well close your pr until you're ready

**Comment by Iamgoofball** [comment]
@optimumtact I wrote that really quickly while rushing to work

This is the PR that is testmerged on the 511 test server so that anyone can hop on and test out 60fps or some shit like that.

**Comment by MrStonedOne** [comment]
I should point out, that a pr doesn't have to be open to be test merged.

**Comment by Iamgoofball** [comment]
it does need to be open to actually update commits however

**Comment by MrStonedOne** [comment]
so you should close it before oranges does so you can open it without having to nag oranges

**Comment by Iamgoofball** [comment]
dream is dead

**Pull request closed without merging**